model-optimizer/extensions/front/global_pooling_to_reduce.py

```diff
@@ -45,8 +45,8 @@ def find_and_replace_pattern(self, graph: Graph):
 
             pooling.out_port(0).get_connection().set_source(reduce.out_port(0))
             src = pooling.in_port(0).get_connection().get_source()
-            pooling.in_port(0).disconnect()
-            src.connect(reduce.in_port(0))
```

Comment:
@popovaan, what if we swap these two lines? Will it be fixed?
How can we prevent the similar issue in the future?

Reply:
src.connect(reduce.in_port(0)) here calls add_destination() that creates new edge without tensor names or other attributes. 
In this transformation we plan to remove pooling node, that is connected to src. So it is better to use set_destination() or set_destination() that moves attributes to the new connection. And not use add_distanation that create empty new edge.

So we could do like this:
reduce.in_port(0).get_connection().set_source(src)
pooling.in_port(0).disconnect()
As we copied attributes to new connection, now we can disconnect the old node. But clean_up() removes pooling node anyway so it is not necessary. 

Or we can do like this:
src.connect(reduce.in_port(0))
// no disconnect
It is a dangerous choice, because attributes are on the edge that is about to be removed. In this case all depends on how other connections are implemented. If there will be set_source()/set_destionation() with src then attributes will be kept. As set_source()/set_destionation() gather attributes from all ingoing/outgoing edges.

In general it is recommended to use set_source()/set_destionation() for nodes that were already in graph. And use connect/add_destination between new created nodes.